src/main/java/us/tastybento/bskyblock/database/flatfile/FlatFileDatabaseHandler.java

```diff
@@ -379,18 +484,30 @@ private Object deserialize(Object value, Class<? extends Object> clazz) {
     @Override
     public void deleteObject(T instance) throws IllegalAccessException, IllegalArgumentException, InvocationTargetException, IntrospectionException {
         // The file name of the Yaml file.
-        PropertyDescriptor propertyDescriptor = new PropertyDescriptor("uniqueId", type);
+        PropertyDescriptor propertyDescriptor = new PropertyDescriptor("uniqueId", dataObject);
         Method method = propertyDescriptor.getReadMethod();
         String fileName = (String) method.invoke(instance);
         if (!fileName.endsWith(".yml")) {
             fileName = fileName + ".yml";
         }
         File dataFolder = new File(plugin.getDataFolder(), DATABASE_FOLDER_NAME);
-        File tableFolder = new File(dataFolder, type.getSimpleName());
+        File tableFolder = new File(dataFolder, dataObject.getSimpleName());
         if (tableFolder.exists()) {
             File file = new File(tableFolder, fileName);
             file.delete();
         }
     }
 
+    /* (non-Javadoc)
+     * @see us.tastybento.bskyblock.database.managers.AbstractDatabaseHandler#loadSettings(java.lang.String, java.lang.Object)
+     */
+    @Override
+    public T loadSettings(String uniqueId, T dbConfig) throws InstantiationException, IllegalAccessException, IllegalArgumentException, InvocationTargetException, ClassNotFoundException, IntrospectionException {
+        if (dbConfig == null) return loadObject(uniqueId);
+        
+        // TODO: compare the loaded with the database copy
```

Comment:
This is where the annotations needsReset = true and overrideOnChange = true need to be processed.